### PR TITLE
executor: fix wrong enum key in point get

### LINKED ISSUE #24562: batch point get for enum type get wrong result
## Bug Report

Please answer these questions before submitting your issue. Thanks!

### 1. Minimal reproduce step (Required)
```sql
CREATE TABLE PK_TCOLLATION3966STROBJSTROBJ (COL1 ENUM('^]IVWXEYONSORLSLQK3D_MZVT1?QT>HRUTLHLBTNPQPQUY@09<8N=YKIZ[RFA2K:V<S:RCVZGZG9_2K8H^O6ZSUU','DUQFI9QLXFF2CDUXN2KYH`D_FTOJA=M4SJLOEN0CG;F60RVSBZPBJM','UXHE3QX5WP`HM4LTWAALA[^MTHK:UP?]?Q7Q;ZEDOGXS^4[JSV6`4RO','MNRCTDNKGYFX=26\PZ[=OXG6KLKD5PRRNG`5CFTQRB\N<AIOB?WFEVX2ZAJXI;HI:DNIF[PJN^DU`1LWD9QNTUL^CH4;=86BKD?VCQGDOBRQ','H]HF6MPWZI0NPV@K^','U4G4QRNNFKE?P9Z[R^=BIM]C\RFNGDOBPMZMTMJHC3OEW;<WF@BW2GAOZVQJW__DB[6E[O\Q9TE\4M<','WLIQMCNAI^HSZUH8C2R^A^7OAPWG2VYACCCVH3[<XPXNAXM\PKDNVL_MCE4;2`C^AE0WFSLO_18@;O\3VHXRWCK`LP?PVZJQE4FKH8QA=U]I4XOBR5LWYEF=>XYVZ','MOKBNX71_@ADC4X9L@]CUVQNR]76WV1INCYVEKO9XMAIV^S=KNU_2L;_\3JB48_W^BP=GQ_H=W\C9T03>KPPOFYN[50GSLE\DP6XX=X','173IZBDJKTTR6LNCTOWWJDCG\K;PA]UVB4P[\>FHBN7AO7E?RXT<WXDRDB<29XH`GZVHJ97IG>ZYX4HA;BLHP6EB_?_3@ZFV<MZOKWOY>3^Q6]N5C=>CNF^W2L','9YGUQ:MLWDM<B'),COL2 VARCHAR(20),COL4 DATETIME,COL3 BIGINT,COL5 FLOAT,PRIMARY KEY(COL1)) CHARSET UTF8 COLLATE UTF8_GENERAL_CI;

insert into PK_TCOLLATION3966STROBJSTROBJ (COL1,COL2,COL4,COL3,COL5) values(6,"虡摿驟蚔冿峥仕牅癲ȱ矙榓咯腝縌窄軓旨鵁褪","5545\-08\-08\ 14:57:40",6286550474446791307,1.355671759964334e+38); 

select * from PK_TCOLLATION3966STROBJSTROBJ where col1 in ("3XTeGWJCJDEsg1WF6tZ1REChv1rH6SxGOATuBYaPKhgGPeIHVQwkjKf1PtGUVZTg3gHjE4V4QyCIsZzUytwcMufbBe", "6", "4l3lij9");
explain select * from PK_TCOLLATION3966STROBJSTROBJ where col1 in ("3XTeGWJCJDEsg1WF6tZ1REChv1rH6SxGOATuBYaPKhgGPeIHVQwkjKf1PtGUVZTg3gHjE4V4QyCIsZzUytwcMufbBe", "6", "4l3lij9");

```

### 2. What did you expect to see? (Required)
```sql
MySQL > select * from PK_TCOLLATION3966STROBJSTROBJ where col1 in ("3XTeGWJCJDEsg1WF6tZ1REChv1rH6SxGOATuBYaPKhgGPeIHVQwkjKf1PtGUVZTg3gHjE4V4QyCIsZzUytwcMufbBe", "6", "4l3lij9");

+------+------+------+------+------+
| COL1 | COL2 | COL4 | COL3 | COL5 |
+------+------+------+------+------+

```
### 3. What did you see instead (Required)
```sql
MySQL > select * from PK_TCOLLATION3966STROBJSTROBJ where col1 in ("3XTeGWJCJDEsg1WF6tZ1REChv1rH6SxGOATuBYaPKhgGPeIHVQwkjKf1PtGUVZTg3gHjE4V4QyCIsZzUytwcMufbBe", "6", "4l3lij9");
+------------------------------------------------------------------------------+-----------------------------------------+---------------------+---------------------+-------------+
| COL1                                                                         | COL2                                    | COL4                | COL3                | COL5        |
+------------------------------------------------------------------------------+-----------------------------------------+---------------------+---------------------+-------------+
| U4G4QRNNFKE?P9Z[R^=BIM]CRFNGDOBPMZMTMJHC3OEW;<WF@BW2GAOZVQJW__DB[6E[OQ9TE4M< | 虡摿驟蚔冿峥仕牅癲ȱ矙榓咯腝縌窄軓旨鵁褪 | 5545-08-08 14:57:40 | 6286550474446791307 | 1.35567e+38 |
+------------------------------------------------------------------------------+-----------------------------------------+---------------------+---------------------+-------------+

1 row in set
Time: 0.053s
MySQL > explain select * from PK_TCOLLATION3966STROBJSTROBJ where col1 in ("3XTeGWJCJDEsg1WF6tZ1REChv1rH6SxGOATuBYaPKhgGPeIHVQwkjKf1PtGUVZTg3gHjE4V4QyCIsZzUytwcMufbBe", "6", "4l3lij9");
+-------------------+---------+------+--------------------------------------------------------------------+------------------------------+
| id                | estRows | task | access object                                                      | operator info                |
+-------------------+---------+------+--------------------------------------------------------------------+------------------------------+
| Batch_Point_Get_1 | 3.00    | root | table:PK_TCOLLATION3966STROBJSTROBJ, clustered index:PRIMARY(COL1) | keep order:false, desc:false |
+-------------------+---------+------+--------------------------------------------------------------------+------------------------------+


```
### 4. What is your TiDB version? (Required)

```sql
MySQL root@172.16.4.131:hchwang> select tidb_version()\G
***************************[ 1. row ]***************************
tidb_version() | Release Version: v4.0.0-beta.2-2832-gc6c8265e0-dirty
Edition: Community
Git Commit Hash: c6c8265e098b65339a85be2e35678b1c13f0f53a
Git Branch: master
UTC Build Time: 2021-05-10 14:58:41
GoVersion: go1.13
Race Enabled: false
TiKV Min Version: v3.0.0-60965b006877ca7234adaced7890d7b029ed1306
Check Table Before Drop: false
```

### PR DESCRIPTION
**Pull request by lzmhhh123**
Signed-off-by: lzmhhh123 <lzmhhh123@gmail.com>

### What problem does this PR solve?

Issue Number: close #24562

### What is changed and how it works?

How it Works:

### Related changes

- Need to cherry-pick to the release branch

### Check List 

Tests 

- Unit test
- Integration test

### Release note 

- fix wrong enum key in point get

### CODE REVIEW

File: executor/point_get.go [hunk 1]
```diff
@@ -442,6 +442,16 @@ func EncodeUniqueIndexValuesForKey(ctx sessionctx.Context, tblInfo *model.TableI
...
 			str, err = idxVals[i].ToString()
 			idxVals[i].SetString(str, colInfo.FieldType.Collate)
+		} else if colInfo.Tp == mysql.TypeEnum && (idxVals[i].Kind() == types.KindString || idxVals[i].Kind() == types.KindBytes || idxVals[i].Kind() == types.KindBinaryLiteral) {
+			var str string
+			var e types.Enum
+			str, _ = idxVals[i].ToString()
+			e, err = types.ParseEnumName(colInfo.FieldType.Elems, str, colInfo.FieldType.Collate)
+			if err != nil {
+				e = types.Enum{}
+				err = nil
```
Comment (wshwsh12) [inline executor/point_get.go:452]: Why not return nil,kv.ErrNotExist?

**Review by ichn-hu — APPROVED** [review]

**Pull request merged**